tests/test_config.py

```diff
@@ -212,3 +214,30 @@ def test_ini_config_bash():
     cfg.update({'dressing': ['ranch', 'italian']})
     expected['dressing'] = ['ranch', 'italian']
     assert cfg == expected
+
+def test_transform_config():
+    '''Test that transforms work as intended.
+
+    '''
+    for test1, test2 in itertools.permutations(["INI", "YAML", "F90"],2):
+        test1file = "NML" if test1 == "F90" else test1
+        test2file = "NML" if test2 == "F90" else test2
```

Comment:
We had several discussions about this becoming a Rube Goldberg machine. I'm glad to see that we have the tools necessary to make that logic fairly compact and straightforward.

I think it could be helpful to add a quick comment before these two lines to mention that we want to transform Config subclass type to an appropriate file subscript. It took me a little while to understand why we needed this transformation.